### PR TITLE
feat: add `MessageActionRow` components in MessageObject

### PR DESCRIPTION
**Pull request by imranbarbhuiya**
## Changes

This pr adds a new option for `components` in `MessageObject`

## Status

- [x] These changes have been tested and formatted properly.
- [ ] This PR includes only documentation changes (JSDoc, README or typings), no code change.
- [ ] This PR introduces some breaking changes.

### CODE REVIEW

**Comment by Nico105** [comment]
_finger slipped_

**Comment by imranbarbhuiya** [comment]
> Also maybe filling in `linkURL` _or what its name is*_ would be good if someone wants to link to the gaw without editing the message after start()

I've changed the code to allow linkbutton and selectmenu.

**Comment by imranbarbhuiya** [comment]
I'm not that good at wording. Please change it to whatever will suit.

**Comment by Nico105** [comment]
both blocks are referring to `winMessage`...
But anyway, we would need such a block for noWinnerMessage and error of rerroll too.
So lets just make it a `fillInComponents` function, no?

**Comment by imranbarbhuiya** [comment]
> both blocks are referring to `winMessage`...
> But anyway, we would need such a block for noWinnerMessage and error of rerroll too.
> So lets just make it a `fillInComponents` function, no?

I've updated the code to allow no winner messages also.

File: src/Giveaway.js [hunk 1]
```diff
@@ -371,6 +371,30 @@ class Giveaway extends EventEmitter {
         return embed;
     }
 
+    /**
+     * @param {Discord.MessageActionRow[]} components The components that should get filled in.
+     * @returns {Discord.MessageActionRow[]} The filled in components.
+     */
+    fillInComponents(components) {
+        if (!components?.length) return components;
```
Comment (Nico105) [inline src/Giveaway.js:379]: wouldn't returning null like the for the other methods be better?
cause otherwise the returned value might not match the doc

Comment (Nico105) [reply src/Giveaway.js:379]: [MessageOptions](https://github.com/discordjs/discord.js/blob/988a51b7641f8b33cc9387664605ddc02134859d/src/structures/MessagePayload.js#L141)
they don't explicitly state it, but it actually is nullable, same in v14 dev

Comment (imranbarbhuiya) [reply src/Giveaway.js]: fixed

**Comment by Nico105** [comment]
we would have to add [`| MessageActionRowOptions`](https://discord.js.org/#/docs/discord.js/stable/typedef/MessageActionRowOptions) too right?
Cause it's at least guaranteed through serialization in case of winMessage.

**Comment by imranbarbhuiya** [comment]
> we would have to add [`| MessageActionRowOptions`](https://discord.js.org/#/docs/discord.js/stable/typedef/MessageActionRowOptions) too right? Cause it's at least guaranteed through serialization in case of winMessage. (function behaviour doesn't change)

added

**Review by Nico105 — APPROVED** [review]
should be fine right?

**Comment by imranbarbhuiya** [comment]
> should be fine right?

Yeah.

**Comment by Nico105** [comment]
👍

**Pull request merged**